loader/loader.go

```diff
@@ -204,36 +204,53 @@ syntax = "proto3";
 	return fset.File, nil
 }
 
-// SplitGunkTag splits a '+gunk' tag from a comment group, returning the leading
-// documentation and the tag's Go expression.
-func SplitGunkTag(fset *token.FileSet, comment *ast.CommentGroup) (string, ast.Expr, error) {
+// SplitGunkTag splits '+gunk' tags from a comment group, returning the leading
+// documentation and the tags Go expressions.
+func SplitGunkTag(fset *token.FileSet, comment *ast.CommentGroup) (string, []ast.Expr, error) {
+	// Remove the comment leading and / or trailing identifier; // and /* */ and `
 	docLines := strings.Split(comment.Text(), "\n")
-	var tagLines []string
+	var gunkTagLines []string
+	var gunkTagPos []int
+	var commentLines []string
+	// If the +gunk tag is a multiline tag (contains a '{'), then
+	// we need to wait until we find the closing '}'.
+	waitingForMoreLines := false
```

Comment:
I'm not sure I understand why we need this. Why not simply stop each `+gunk` tag at the next `+gunk` line, or at the end of the comments?

Another advantage is that we wouldn't break in cases like:

```
+gunk foo.Bar{
    F: foo.Baz{...}
}
```

Reply:
A better way to implement your heuristic would be to count opening and closing braces, stopping when we reach 0. Though then we of course have the issue of `+gunk foo.Bar("{")`.

Reply:
I went with this approach because I assumed you could have comments in-between gunk tag? If that isn't possible then I agree that your approach is much cleaner.

Reply:
I think not allowing comments in between or after gunk tags is our best short-term strategy. If we want to support those comments in the future, there are ways to do that that would be more reliable than string-based heuristics like the one you implemented. But they'd be tricky to implement and would easily require fifty lines of non-trivial code, so I just don't think it's worth it right now.